dbrequests/mysql/database.py

```diff
@@ -78,6 +78,31 @@ def send_data(self, df, table, mode='insert', **params):
         with self.transaction() as conn:
             return conn.send_data(df, table, mode, **params)
 
+    def send_delete(self, df, table: str, mode: str = 'in_set', **params) -> int:
+        """
+        Delete entries in a table. Use this method instead of send_bulk_query
+        for deleting rows in a table; e.g. instead of looping over a long
+        vector and deleting chunkwise.
+
+        - df (DataFrame): a data frame containing the information which rows to
+          delete. See mode for details.
+        - table (str): the table where rows are to be deleted
+        - mode (str):
+          - 'in_join': delete entries with a match in df. If it is possible to
+            do a left join with df we have a match.
+          - 'not_in_join': delete entries with **no** match in df.
+          - 'in_set': delete entries which are in the set defined by df. We do
+            a 'where in col from df' and concatenate columns with 'and'. This
+            can bring a considerable speedup, compared to the join strategy, if
+            you only delete values from one enum field.
+          - 'not_in_set': delete entries which are not in the set defined by
+            df.
```

Comment:
Aaah, und "(not_)in_set" implementiert unser bisheriges `delete_where`?

Reply:
Ja. Alternativ auch [not_]in_join. Es gibt dabei Szenarien, in denen die eine Variante deutlich schneller ist als die andere. Wenn man einfach eine Lange Liste von IDs hat, dann macht es fast keinen Unterschied. Wenn man auf zwei Kategorien eines enums filtert, dann ist in_cols deutlich schneller. Wenn man auf eine lange Liste von IDs filtert und zusätzlich noch auf eine nicht indizierte Spalte, dann ist der join schneller. Schneller bedeutet jeweils, dass ich bei 20 Mio Zeilen gewartet habe, dass es durchläuft (~5Min). Und langsam, dass es auch nach ~20Min noch kein Ergebnis gab. Es gibt also deutliche Unterschiede. Fazit: Löschen ist teuer.